fdbserver/Status.actor.cpp

```diff
@@ -874,9 +874,8 @@ ACTOR static Future<JsonBuilderObject> processStatusFetcher(
 	}
 
 	state std::vector<OldTLogConf>::const_iterator oldTLogIter;
-	for (oldTLogIter = db->get().logSystemConfig.oldTLogs.begin();
-	     oldTLogIter != db->get().logSystemConfig.oldTLogs.end();
-	     ++oldTLogIter) {
+	state std::vector<OldTLogConf> oldTLogs = db->get().logSystemConfig.oldTLogs;
```

Comment:
LGTM. Some notes:

- Since you're making a copy, I am assuming it's low cost and there are not too many elements in the vector.
- Also, is there a way to make oldTLogs const? Flow won't allow me. But it'd be nice to do it, because it's possible someone adds code later by mistake that mutates oldTLogs in the loop.

Reply:
Flow doesn't allow this to be `const`, because of the assignment. Usually the number of old generation is 0, so this is a low cost operation most of the time.

Reply:
> Flow doesn't allow this to be const, because of the assignment.

In general, how do we prevent people from mutating variables that were supposed to be const? Documentation or is there a better flow way?

Reply:
We don't have a good way. Proper documentation, e,g., comments, are the best we can do.

The implicit rule in flow is that before any `wait()` statement, no other co-routines can interrupt the execution and modify variables.

Reply:
```cpp
state std::vector<OldTLogConf>::const_iterator oldTLogIter;
	const std::vector<OldTLogConf> oldTLogsV = db->get().logSystemConfig.oldTLogs;
	state const std::vector<OldTLogConf>* oldTLogs = &oldTLogsV;
	for (oldTLogIter = oldTLogs->begin(); oldTLogIter != oldTLogs->end(); ++oldTLogIter) {
		for (auto& tLogSet : oldTLogIter->tLogs) {
			for (auto& it : tLogSet.tLogs) {
				if (it.present()) {
					roles.addRole("log", it.interf());
				}
			}
			for (auto& it : tLogSet.logRouters) {
				if (it.present()) {
					roles.addRole("router", it.interf());
				}
			}
		}
		wait(yield());
	}
```

That's one hack, although you're dealing with pointers and more code so not as easy to read. 

But now you can't do `oldTLogs->clear()` but can still do `oldTlogs = nullptr` since we can't do `const std::vector<OldTLogConf>* const oldTLogs` for the same Flow reason.